app/javascript/packages/request/index.ts

```diff
@@ -0,0 +1,40 @@
+interface RequestOptions extends RequestInit {
+  /**
+   * Either boolean or unstringified POJO to send with the request as JSON. Defaults to true.
+   */
+  json?: object | boolean;
+
+  /**
+   * Whether to include CSRF token in the request. Defaults to true.
+   */
+  csrf?: boolean;
+}
+
+const isPOJO = (obj) => Object.prototype.toString.call(obj) === '[object Object]';
+
+const getCSRFToken = () =>
+  document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
+
+export const request = async (url: string, options: Partial<RequestOptions> = {}) => {
+  const { csrf = true, json = true, ...fetchOptions } = options;
+  let { body, headers } = fetchOptions;
+  headers = new Headers(headers);
+
+  if (csrf) {
+    const csrfToken = getCSRFToken();
+    if (csrfToken) {
+      headers.set('X-CSRF-Token', csrfToken);
+    }
+  }
+
+  if (json) {
+    headers.set('Content-Type', 'application/json');
+
+    if (isPOJO(json)) {
```

Comment:
Would it be simpler to check that it's not the boolean type?

```suggestion
    if (typeof json !== 'boolean') {
```

I suppose the utility method could make it clearer why we're JSONifying it, though the actual implementation of checking the object type is rather complex, comparatively speaking. (It also may not account for certain edge cases like if someone would want to pass an object created through `Object.create(null)`, which can occasionally be a useful way to create an object without its default prototype members)

Reply:
this makes me want to troll the api with `{ json: new Boolean(true) }` 😛